homeassistant/components/dsmr_reader/sensor.py

```diff
@@ -69,7 +69,14 @@ async def async_added_to_hass(self) -> None:
         @callback
         def message_received(message):
             """Handle new MQTT messages."""
-            if self.entity_description.state is not None:
+            if message.payload == "":
+                # On empty payload, reset the sensor value or ignore update
+                if self._attr_native_value is not None:
+                    self._attr_native_value = None
+                else:
+                    return
```

Comment:
This can be simplified to just this:

```suggestion
                self._attr_native_value = None
```

The state can always be written, it won't cause changes in case the state is already the same (unless forced, which doesn't seem to be the case here).

../Frenck